src/main/config/messages_template.xml

```diff
@@ -7,413 +7,415 @@ Copyright 2005, 2006 IBM Corporation
   </message>
-  
+
   <message id="DOTJ048I" type="INFO">
-    <reason>Unable to find key definition for key reference "%1" in scope "%2".</reason>
-    <response>The href attribute may be used as fallback if it exists</response>
+    <reason>Unable to find key definition for key reference '%1' in scope '%2'.</reason>
+    <response>Using the @href attribute as fallback if it exists.</response>
   </message>
-  
+
   <message id="DOTJ049W" type="WARN">
-    <reason>The attribute value %1="%3" on element "%2" does not comply with the specified subject scheme.</reason>
-    <response>According to the subject scheme map, the following values are valid for the %1 attribute: %4</response>
+    <reason>The @%1 attribute value '%3' on the &lt;%2&gt; element does not comply with the specified subject scheme.</reason>
+    <response>According to the subject scheme map, the following values are valid for the @%1 attribute: '%4'.</response>
   </message>
-  
+
   <message id="DOTJ050W" type="WARN">
     <reason>Found an &lt;index-see&gt; or &lt;index-see-also&gt; reference to the term '%1', but that term is not defined in the index.</reason>
     <response></response>
   </message>
-  
+
   <!-- Long term, would like to split this into two messages. One for file not found, another for file outside the scope of the map. -->
   <message id="DOTJ051E" type="ERROR">
-    <reason>Unable to load target for coderef "%1".</reason>
-    <response></response>
+    <reason>Unable to load target for coderef '%1'.</reason>
+    <response>Make sure '%1' exists and that you have permission to access it.</response>
   </message>
 
   <message id="DOTJ052E" type="ERROR">
-    <reason>Code reference charset "%1" not supported.</reason>
-    <response>See the DITA-OT documentation for supported charset values on the format attribute.</response>
+    <reason>Unsupported code reference charset '%1'.</reason>
+    <response>See the documentation for supported character set values on the @format attribute.</response>
   </message>
 
   <!-- Obsolete since 2.3 -->
   <message id="DOTJ053W" type="WARN">
-    <reason>Input file '%1' is not valid DITA file name.</reason>
-    <response>Please check '%1' to see if it is correct. The extensions ".dita" or ".xml" are supported for DITA topics.</response>
+    <reason>The input resource '%1' is not a valid DITA filename.</reason>
+    <response>Check '%1' to see if it is correct. The '.dita' and '.xml' file name extensions are supported for DITA topics.</response>
   </message>
-  
+
   <message id="DOTJ054E" type="ERROR">
-    <reason>Unable to parse invalid %1 attribute value "%2"</reason>
+    <reason>Unable to parse invalid @%1 attribute value '%2'.</reason>
     <response></response>
   </message>
-  
+
   <message id="DOTJ055E" type="ERROR">
-    <reason>Invalid key name "%1".</reason>
-    <response></response>
+    <reason>Invalid key name '%1'.</reason>
+    <response>Check the reference to see if the target exists.</response>
```

Comment:
I think the response here is wrong - the message is not saying that the target is missing, the message is saying "the value on `@keys` is not valid". For example it will fail if you use `keys="{bad?key}"` with the special character list here https://github.com/dita-ot/dita-ot/blob/6e5d9ba7d7ce842852d2d6100372deec62360874/src/main/java/org/dita/dost/writer/ValidationFilter.java#L369

Reply:
Added the list of disallowed characters.